ledger/participant-integration-api/src/main/scala/platform/store/interfaces/TransactionLogUpdate.scala

```diff
@@ -29,6 +29,7 @@ object TransactionLogUpdate {
     */
   final case class Transaction(
       transactionId: String,
+      commandId: Option[String],
```

Comment:
Any reason why this is an `Option`? If it is meant for a two-phase population of `Transaction`, I'd suggest just populating it from the get-go in `InMemoryStateUpdater` and filter it in `TransactionLogUpdatesConversion`

Reply:
As far as I understand to get the `commandId` we must first have filtered the events, so we cannot populate it in  `InMemoryStateUpdater`. I could use just a simple `String` and set it to be the empty string. Would that be better?

Reply:
There are two different concepts at play here. Let's clarify: 
* The `commandId` is a command/transaction specific field that is set by the client on a command submission. Furthermore, this field is also available in `Update.TransactionAccepted` coming from the `ReadService` if the original command submitter is local to the participant (i.e. the party is allocated on the participant). 
* When we construct transaction streams for the clients, we populate the `commandId` if any of the requesting parties is an original submitter of the command. Otherwise, we forward empty. I believe this is needed for privacy considerations (i.e. in order to not _leak_ informations about the original submission to other parties that did not request it.)

Reply:
Updated the PR with: https://github.com/digital-asset/daml/pull/14447/commits/65fc67f1d204bd4d64b453b9a3ebd45fc7f21bdb